scripts/common/get_logs.sh

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+
+__author__="Amandeep Singh Arora"
+__version__="1.0"
+
+> /tmp/fledge_syslog.log
+
+exec 2<&-
+exec 2<>/tmp/fledge_syslog.log
+
+RECALC_AFTER_N_SCRIPT_RUNS=100
+NUM_LOGFILE_LINES_TO_CHECK_INITIALLY=2000
+
+offset=100
+limit=5
+pattern=""
+level="debug"
+logfile="/var/log/syslog"
+sourceApp="fledge"
+
+while [ "$#" -gt 0 ]; do
+  case "$1" in
+    -pattern) pattern="$2"; shift 2;;
+    -offset) offset="$2"; shift 2;;
+    -limit) limit="$2"; shift 2;;
+    -level) level="$2"; shift 2;;
+    -source) sourceApp="$2"; shift 2;;
+    -logfile) logfile="$2"; shift 2;;
+  esac
+done
+
+sum=$(($offset + $limit))
+echo "" >&2
+echo "****************************************************************************************" >&2
+echo "************************************* START ********************************************" >&2
+
+if [ -f /tmp/fledge_syslog_script_runs ]; then
```

Comment:
There is a danger here if we have two users logged in and looking at syslog

Reply:
Since these requests go thru' fledge core, would the requests from 2 browsers be processed sequentially by core or they could interleave. Even if they interleave, the script_runs count would increase doubly fast, which shouldn't be a problem. Or do you think a worse race condition could occur in this case?

Reply:
We can run multiple threads to make API requests in theory. It may be that Python GIL actively prevents this, but possibly not as I belie these commands are run using asyncio mechanisms.